src/checkpoint/checkpoint_txn.c

```diff
@@ -1333,6 +1354,18 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
     WT_STAT_CONN_SET(session, checkpoint_state, WT_CHECKPOINT_STATE_COMMIT);
     WT_ERR(__wt_txn_commit(session, NULL));
 
+    /*
+     * Crash if the checkpoint crash feature is enabled and configured to fail before updating the
+     * metadata.
+     */
+    if (ckpt_crash_before_metadata_upd) {
+#ifdef _WIN32
```

Comment:
Suggestion: It would be cleaner to turn it into a function (maybe somewhere in the os code headers)? Not sure what would be a good name for it; maybe `__wt_debug_crash()` or something like that.

Reply:
Fixed in 9d64d04ec275d9d111c19e36ac8584074dda0211